### ISSUE TITLE
"rails server" command not working

### ISSUE THREAD

**Issue by arielkirkwood** [colindean/plas#38]
Hello,

I'm relatively new to Github so I'm not sure if this is the best way to write down this issue, but I wanted to try out PLAS, so I followed all the steps:
- got RVM set up with three rubies: 1.8.7, 1.9.2, and ruby-head
- git cloned PLAS into home directory on a Mac
- cd'd to the directory, accepted rvm's trust prompt for the .rvmrc file
- bundle install worked as expected
- rake db:migrate worked as expected
- rails server came up with the following error:
  https://gist.github.com/982397

Again, sorry if I could've explained this better. I'd be happy to share any more information you might need to fix this issue.

**Comment by arielkirkwood** [comment]
I tried creating a new skeleton project with rails, and it works as expected, I can navigate to localhost:3000 and I see the default splash page.

**Comment by colindean** [comment]
Thanks for your interest in PLAS! Can you give me the output of `gem list --local`? It looks like it's dying before it even gets to real PLAS code. Our version of bundler is different. 

That it works on a newly generated app is odd, and certainly points to PLAS or one of its Gem dependencies. If you create a new app and use PLAS's Gemfile, does the server still start?

**Comment by arielkirkwood** [comment]
I'll definitely try out that command when I get home from work.

How might I try your second suggestion? I'm pretty new to ruby and rails, I haven't actually learned the language yet :P I'm more familiar with PHP and CakePHP at the moment, but I love LAN parties, and when I found this, I had to try it out.

**Comment by colindean** [comment]
The best way to use PLAS's Gemfile is simply to copy it into the newly created rails app. There's nothing else required, really.

I wouldn't be surprised if some of the sources in the PLAS gemfile are out of date. I used to develop against ruby-head, but got tired of hitting problems with gems and started using 1.9.2. A few of the gems have sources which are my own gh forks or forks from other people fixing weird ruby-head interoperability problems or rails 3.0.0 problems during beta. That's what I get for using not one but TWO edge repos. Now, with Rails 3.1 coming out, there's a few more things to look at fixing...

**Comment by arielkirkwood** [comment]
Here's my local gems and my gemsets: https://gist.github.com/985565

And, I tried copying the gemfile and gemfile.lock files to a new Rails app, here's the result when I did `bundle install` and then tried to run `rails server`: https://gist.github.com/985591

**Comment by colindean** [comment]
Yeah, that's a problem with ffi. Take off the version specifier in the Gemfile and see what happens.

Or, you _should_ be able to remove that entire block from the Gemfile (see the comments around it). Those were necessary for getting those Gems working with ruby-head on Ubuntu Natty Alpha 2. Clearly, that's a long time ago, so they shouldn't be needed any more.

**Comment by arielkirkwood** [comment]
I removed the line for ffi in that block of the Gemfile, here was my result for `rails server`: https://gist.github.com/986176
I also tried removing the entire block from the Gemfile, and this was the result: https://gist.github.com/986085

I noticed that `which ruby` always shows this after I run `bundle install` for PLAS:
/Users/arielkirkwood/.rvm/rubies/ruby-1.9.2-p180/bin/ruby

I am able to `rvm use ruby-head` and it sticks for trying things, but during the `bundle install` process, I get this error, regardless of what I do to the Ubuntu Natty portion of the Gemfile: https://gist.github.com/986177

Trying `rails server` after that error: https://gist.github.com/986179

Current workflow, just fyi:
`git clone git@github.com:neonelectro/plas`
`cd plas`
`nano Gemfile` (to edit the Gemfile)
`rvm gem install bundler`
`bundle install`
`rails server`

Hope I'm helping.

**Comment by colindean** [comment]
I'm still pointing the finger to ffi. 

I noticed that you're running 10.7, developer preview? Perhaps there's something weird about ffi on 10.7?

Can you create a blank rails project and put `gem "ffi"` in the Gemfile? Ensure that you're using a bare gemset, too (create a new one and `gem install bundler` then do `bundle install`).

**Comment by arielkirkwood** [comment]
That's really odd, I'm on a MacBook Pro runnning [10.6.7](http://i.imgur.com/pXgPi.png) :P I'm curious, where did you get that from?

Anyways, here's the [result](https://gist.github.com/991797) of `bundle install` after starting a blank rails project with [stock gemfile + ffi](https://gist.github.com/991801). The `rails server` command worked successfully.

**Comment by colindean** [comment]
It appears from line 3 of one of your gists that you're running 10.7.

```
ruby 1.9.2p180 (2011-02-18 revision 30909) [x86_64-darwin10.7.0]
```

I've got 10.6.7, as well, and mine shows as:

```
ruby 1.9.2p180 (2011-02-18 revision 30909) [x86_64-darwin10.6.0]
```

Do you have Xcode 4 installed? I don't know why the OS versions would be different. Can you give me the output of `ruby --version`?

Perhaps if I have time, I'll do a fresh checkout and gemset and see what I can figure out.

**Comment by arielkirkwood** [comment]
Here's the version info I could get using both `ruby` and `rvm`.

`ruby --version`:
    ruby 1.9.3dev (2011-05-20 trunk 31646) [x86_64-darwin10.7.0]

`rvm --version`:
    rvm 1.6.10 by Wayne E. Seguin (wayneeseguin@gmail.com) [https://rvm.beginrescueend.com/]

`rvm ruby --version`:
    ruby 1.8.7 (2011-02-18 patchlevel 334) [i686-darwin10.7.0]
    ruby 1.9.2p180 (2011-02-18 revision 30909) [x86_64-darwin10.7.0]
    ruby 1.9.3dev (2011-05-20 trunk 31646) [x86_64-darwin10.7.0]

Thanks for helping out with this, I really appreciate it! I'm quite new to Ruby, but if I start picking it up, I'd be happy to help contribute more to this project if you needed help. I'm sort of starting a similar project using PHP and CakePHP, called [LANbot](https://github.com/neonelectro/LANbot).

**Comment by arielkirkwood** [comment]
I don't have Xcode 4 installed, but I'm pretty sure I installed the dev tools at one point... I might not even have that :( Let's just assume I don't.

EDIT: If I go buy Xcode 4 from the Mac App store right now, will that help? I don't mind spending $5, and I'd been meaning to check it out, anyways... (or I could pirate it to test it out)

**Comment by colindean** [comment]
We'd love to have your help! Getting a ruby development environment set up is usually one of the hardest parts of developing in ruby. Ruby is very focused on its gem system, especially for rails development!

I have a theory. I've not recompiled my 1.9.2 since 10.6.7 came out. `uname -a` shows that 10.6.7 is running Darwin kernel 10.7. I'll guarantee that's related! I wonder if there's something weird going on there.

**Comment by colindean** [comment]
As of 660063c9919b63cef506, I updated the Gemfile and made sure it worked at least for a few pages. Bomb your repo and refork, assuming you've made no changes, or just clone from mine and see if you can get it to work.

**Comment by colindean** [comment]
I lied. Get db5d95f58008ebc66b572d119c5354dc7411d5ef, because I messed up and forgot the acts_as_tree gem in rubygems is for rails2 and there's a separate repo for rails3 version.

**Comment by arielkirkwood** [comment]
IT'S FINALLY WORKING!

I thought I'd just try the full nuclear option and re-install RVM, so I did, this time, only installing the ruby 1.9.2-p180 option.
After following the steps in your Contributing doc, I finally tried `rails server` and it worked! Now to check out the site and find bugs and things :)

Edit: To get plas again, I re-forked your latest update, I didn't have to choose a specific commit of yours for it to work.
